tests/test_shell_launcher.py

```diff
@@ -0,0 +1,323 @@
+    with (
+        open(out_file, "w", encoding="utf-8") as out,
+        open(err_file, "w", encoding="utf-8") as err,
+    ):
+        cmd = ShellLauncherCommand(
+            {}, run_dir, out, err, EchoHelloWorldEntity().as_program_arguments()
+        )
+        for _ in range(5):
+            id = shell_launcher.start(cmd)
+            proc = shell_launcher._launched[id]
+            proc.wait()
+            code = shell_launcher.get_status(id)[id]
+            assert code == JobStatus.COMPLETED
+
+
+@requires_slurm
+def test_shell_launcher_returns_failed_status(shell_launcher, test_dir):
+    """Test tht ShellLauncher returns the status of completed Jobs"""
+    run_dir, out_file, err_file = generate_directory(test_dir)
+    with (
+        open(out_file, "w", encoding="utf-8") as out,
+        open(err_file, "w", encoding="utf-8") as err,
+    ):
+        args = (helpers.expand_exe_path("srun"), "--flag_dne")
+        cmd = ShellLauncherCommand({}, run_dir, out, err, args)
+        for _ in range(5):
+            id = shell_launcher.start(cmd)
+            proc = shell_launcher._launched[id]
+            proc.wait()
+            code = shell_launcher.get_status(id)[id]
+            assert code == JobStatus.FAILED
+
+
+def test_shell_launcher_returns_running_status(shell_launcher, test_dir):
+    """Test tht ShellLauncher returns the status of completed Jobs"""
+    run_dir, out_file, err_file = generate_directory(test_dir)
+    with (
+        open(out_file, "w", encoding="utf-8") as out,
+        open(err_file, "w", encoding="utf-8") as err,
+    ):
+        cmd = ShellLauncherCommand(
+            {}, run_dir, out, err, (helpers.expand_exe_path("sleep"), "5")
+        )
+        for _ in range(5):
+            id = shell_launcher.start(cmd)
+            code = shell_launcher.get_status(id)[id]
+            assert code == JobStatus.RUNNING
+        assert all(proc.wait() == 0 for proc in shell_launcher._launched.values())
+
+
+@pytest.mark.parametrize(
+    "psutil_status,job_status",
+    [
+        pytest.param(psutil.STATUS_RUNNING, JobStatus.RUNNING, id="running"),
+        pytest.param(psutil.STATUS_SLEEPING, JobStatus.RUNNING, id="sleeping"),
+        pytest.param(psutil.STATUS_WAKING, JobStatus.RUNNING, id="waking"),
+        pytest.param(psutil.STATUS_DISK_SLEEP, JobStatus.RUNNING, id="disk_sleep"),
+        pytest.param(psutil.STATUS_DEAD, JobStatus.FAILED, id="dead"),
+        pytest.param(psutil.STATUS_TRACING_STOP, JobStatus.PAUSED, id="tracing_stop"),
+        pytest.param(psutil.STATUS_WAITING, JobStatus.PAUSED, id="waiting"),
+        pytest.param(psutil.STATUS_STOPPED, JobStatus.PAUSED, id="stopped"),
+        pytest.param(psutil.STATUS_LOCKED, JobStatus.PAUSED, id="locked"),
+        pytest.param(psutil.STATUS_PARKED, JobStatus.PAUSED, id="parked"),
+        pytest.param(psutil.STATUS_IDLE, JobStatus.PAUSED, id="idle"),
+        pytest.param(psutil.STATUS_ZOMBIE, JobStatus.COMPLETED, id="zombie"),
```

Comment:
Request here for one more param!
```suggestion
        pytest.param(psutil.STATUS_ZOMBIE, JobStatus.COMPLETED, id="zombie"),
        pytest.param("some brand new unknown status", JobStatus.UNKNOWN, id="unknown")
```